### PR TITLE
feat(governance): UPToken and UPSwap deployment script

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This adds scripts to deploy UP token and UP swap contracts easily.

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: governance/hardhat.config.js [hunk 1]
```diff
@@ -70,6 +70,7 @@ const config = {
     compilers: [
       // required to compile oracle as Uniswap v3-periphery supports only solc < 0.8
       { version: '0.7.6', settings: { ...settings, evmVersion: 'istanbul' } },
+      { version: '0.8.20', settings },
```
Comment (clemsos) [inline governance/hardhat.config.js:73]:
```suggestion
```

File: governance/scripts/deployments/emptyProxy.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+const { ethers } = require('hardhat')
```
Comment (julien51) [inline governance/scripts/deployments/emptyProxy.js:1]: Please rename that file to `emptyProxyAndUpSwapImplementation.js`

File: governance/scripts/deployments/emptyProxy.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  const { address: proxy } = await deployUpgradeableContract(EmptyImpl)
+
+  // 2. deploys the UP swap implementation
+  console.log(`Deploying UPSwap implementation...`)
+  const [upSwapQualifiedPath] = await copyAndBuildContractsAtVersion(
+    __dirname,
+    [{ contractName: 'UPSwap', subfolder: 'UP' }]
+  )
+  const { address: impl } = await deployContract(upSwapQualifiedPath)
+
```
Comment (julien51) [inline governance/scripts/deployments/emptyProxy.js:25]:
```suggestion

console.log('Please submit a proposal to the DAO that would upgrade the proxy to the right implementation')
```

File: governance/scripts/deployments/up.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+const { ethers } = require('hardhat')
+const {
+  copyAndBuildContractsAtVersion,
+  deployUpgradeableContract,
+} = require('@unlock-protocol/hardhat-helpers')
+
+async function main() {
```
Comment (julien51) [inline governance/scripts/deployments/up.js:7]:
```suggestion
// This script will be called by the BVI Foundation UP issuer.
async function main() {
```

File: packages/contracts/src/contracts/UP/UPToken.sol [hunk 5]
```diff
@@ -1,4 +1,4 @@
-// Sources flattened with hardhat v2.22.5 https://hardhat.org
+// Sources flattened with hardhat v2.22.3 https://hardhat.org
```
Comment (julien51) [inline packages/contracts/src/contracts/UP/UPToken.sol:1]: ??

File: packages/contracts/src/contracts/UP/UPSwap.sol [hunk 6]
```diff
@@ -1,4 +1,4 @@
-// Sources flattened with hardhat v2.22.5 https://hardhat.org
+// Sources flattened with hardhat v2.22.3 https://hardhat.org
```
Comment (julien51) [inline packages/contracts/src/contracts/UP/UPSwap.sol:1]: ??

**Review by julien51 — COMMENTED** [review]
ok I think we're good.

File: governance/scripts/deployments/emptyProxy.js (hunk 2 above)
Comment (clemsos) [reply governance/scripts/deployments/emptyProxy.js]: ok done

**Review by julien51 — APPROVED** [review]

**Pull request merged**